### PR TITLE
Bugfixing

### PR DESCRIPTION
**Pull request by EmilLa1**
## Purpose

see changelog

### CODE REVIEW

File: src-tauri/Cargo.toml [hunk 1]
```diff
@@ -1,10 +1,9 @@
 [package]
 name = "concordium-lc1c"
-version = "0.0.0"
-description = "A Tauri App"
-authors = ["you"]
-license = ""
-repository = ""
+version = "0.1.3"
+description = "Concordium Localchain"
```
Comment (abizjak) [inline src-tauri/Cargo.toml:4]: Be a bit more descriptive in the description :)

File: src-tauri/LICENSE [hunk 2]
```diff
@@ -0,0 +1,661 @@
+                    GNU AFFERO GENERAL PUBLIC LICENSE
```
Comment (abizjak) [inline src-tauri/LICENSE:1]: We don't really use this licence. We typically have Apache 2 for these kinds of things. So please stick to that.

Also there is no need to have the LICENSE file. Just state the license in Cargo.toml file like for the company id tool.

File: src-tauri/src/common.rs [hunk 3]
```diff
@@ -92,22 +50,24 @@ pub fn create_next_chain_folder(base_path: &Path) -> Result<PathBuf, String> {
 
         if !folder_path.exists() {
             // Create the folder and then return its path
-            std::fs::create_dir(&folder_path).map_err(|e| e.to_string())?;
+            println!("Created chain-{}", counter);
```
Comment (abizjak) [inline src-tauri/src/common.rs:53]: Why stdout? These sorts of things are usually better printed to stderr.

File: src-tauri/src/common.rs [hunk 4]
```diff
@@ -134,135 +94,99 @@ pub async fn instance_list(hash: BlockHash) -> anyhow::Result<HashMap<String, In
...
+    Ok(instances_map)
 }
 
-pub async fn transaction_info(number: AbsoluteBlockHeight) -> anyhow::Result<TransactionsInfo> {
-    println!("Fetching transaction info for block height: {:?}", number);
+pub async fn transaction_info(
+    client: &mut v2::Client,
+    block_hash: BlockHash,
+) -> anyhow::Result<Option<TransactionsInfo>> {
+    println!("Fetching transactions for block: {:?}", block_hash);
```
Comment (abizjak) [inline src-tauri/src/common.rs:116]: stderr I think.

File: src-tauri/src/common.rs [hunk 5]
```diff
@@ -134,135 +94,99 @@ pub async fn instance_list(hash: BlockHash) -> anyhow::Result<HashMap<String, In
...
 
-    Ok(TransactionsInfo {
-        transactions: summaries.to_vec(),
-    })
+    if new_summaries.is_empty() {
+        Ok(None)
+    } else {
+        Ok(Some(TransactionsInfo {
+            transactions: new_summaries.to_vec(),
+        }))
```
Comment (abizjak) [inline src-tauri/src/common.rs:146]: Why do we need Option at all? You can just return transactions = Vec::new()

File: src-tauri/src/common.rs [hunk 6]
```diff
@@ -134,135 +94,99 @@ pub async fn instance_list(hash: BlockHash) -> anyhow::Result<HashMap<String, In
...
-                let amt = info.response.account_amount;
-                amounts_map.insert(a?, amt);
-            }
-            Err(e) => {
-                println!("Failed to get account address. Error: {:?}", e);
-                return Err(anyhow::anyhow!("Failed to get account address: {}", e));
+    // FIXME: Load changes in a more fine-grained manner.
+    if events.next().await.is_some() {
+        state_changed = true;
+    }
```
Comment (abizjak) [inline src-tauri/src/common.rs:161]: YOu should just use get_block_info and check number of transactions in that. It's better than the current check performance wise.

File: src-tauri/src/common.rs [hunk 7]
```diff
@@ -134,135 +94,99 @@ pub async fn instance_list(hash: BlockHash) -> anyhow::Result<HashMap<String, In
...
-    let mut client = v2::Client::new(endpoint_node)
-        .await
-        .context("failed to create client")?;
-    let bi = client
-        .get_block_info(&v2::BlockIdentifier::LastFinal)
+pub async fn listen_finalized_blocks(
+    client: &mut v2::Client,
+) -> anyhow::Result<FinalizedBlocksStream> {
+    client
+        .get_finalized_blocks_from(0u64.into())
```
Comment (abizjak) [inline src-tauri/src/common.rs:189]: Why is this even a function/ It just calls another function with nothing added.

File: src/components/expert-form.tsx [hunk 8]
```diff
@@ -8,14 +8,80 @@ import {
...
+        </CModalFooter>
+      </CModal>
+    </>
+  );
+};
+
+export default ExpertSettingsPage;
+
+export const GENESIS_CONFIGURATION = `
+protocolVersion = "6"
```
Comment (abizjak) [inline src/components/expert-form.tsx:84]: Why is this still embedded in the typescript file?

Comment (abizjak) [reply src/components/expert-form.tsx:84]: You can source it from the toml file.

File: src-tauri/src/views.rs [hunk 9]
```diff
@@ -1,43 +1,85 @@
...
 use tauri::Window;
-use tokio::io::{AsyncBufReadExt, BufReader};
 use tokio::process::Command as AsyncCommand;
+use tokio::task;
+use tokio::time::Duration;
+
+///Software version
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+/// Path to the genesis configuration.
+const GENESIS_CONFIGURATION: &[u8; 4100] = include_bytes!("../../etc/genesis.toml");
```
Comment (abizjak) [inline src-tauri/src/views.rs:35]:
```suggestion
const GENESIS_CONFIGURATION: &[u8] = include_bytes!("../../etc/genesis.toml");
```

File: src-tauri/src/common.rs (hunk 6 above)
Comment (EmilLa1) [reply src-tauri/src/common.rs]: Indeed yes

**Review by abizjak — APPROVED** [review]

**Pull request merged**